### PR TITLE
Add permission check for file download

### PR DESCRIPTION
**Pull request by alexandru-uta**

### CODE REVIEW

File: backend/src/main.rs [hunk 1]
```diff
@@ -73,8 +73,7 @@ fn request_file(request_name: String) -> String {
     with_state_mut(|s| backend::api::request_file(caller(), request_name, s))
 }
 
-#[query]
-fn download_file(file_id: u64) -> FileData {
+fn get_file_data(file_id: u64) -> FileData {
     with_state(|s| match s.file_data.get(&file_id) {
         None => FileData::NotFoundFile,
```
Comment (ielashi) [inline backend/src/main.rs:78]: Wouldn't this case never happen? In what case will we get a `NotFoundFile`?

File: backend/src/main.rs [hunk 2]
```diff
@@ -87,4 +86,15 @@ fn download_file(file_id: u64) -> FileData {
     })
 }
 
+#[query]
+fn download_file(file_id: u64) -> FileData {
```
Comment (ielashi) [inline backend/src/main.rs:90]: test cases please :)

Comment (alexandru-uta) [reply backend/src/main.rs:67]: please check them out

File: backend/src/api/download_file.rs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        request_file(Principal::anonymous(), "request", &mut state);
+
+        // Upload the file, which we assume to have a file ID of zero.
+        let file_id = 0;
+        let _alias = upload_file(file_id, vec![1, 2, 3], &mut state);
+
+        let file_contents: FileData = download_file(&state, file_id, Principal::anonymous());
+        match file_contents {
+            FileData::FoundFile(data) => assert_eq!(data, vec![1, 2, 3]),
+            _ => assert!(false),
```
Comment (ielashi) [inline backend/src/api/download_file.rs:85]:
```suggestion
        assert_eq!(download_file(&state, file_id, Principal::anonymous()), FileData::FoundFile(vec![1,2,3]));
```

**Review by ielashi — APPROVED** [review]

**Pull request merged**